Sources/StreamChat/Repositories/SyncRepository.swift

```diff
@@ -200,6 +204,17 @@ class SyncRepository {
         isRecovery: Bool,
         completion: @escaping (Result<[ChannelId], SyncError>) -> Void
     ) {
+        guard lastSyncAt.numberOfDaysUntilNow < Constants.maximumDaysSinceLastSync else {
+            updateLastSyncAt(with: Date()) { error in
+                if let error = error {
+                    completion(.failure(error))
+                } else {
+                    completion(.success([]))
+                }
+            }
+            return
```

Comment:
Shouldn't we clear the DB in this case?

Reply:
The logic to clear things is at the end of the whole Sync flow.
Whenever there are channels that are not synched, nor watched, those are removed.
If instead we refetch them as part of the sync process, those are kept.